blueprints/disruption-budgets/README.md

```diff
@@ -0,0 +1,208 @@
+# Karpenter Blueprint: NodePool Disruption Budgets
+
+## Purpose
+
+A Karpenter’s NodePool disruption budgets can be used to rate limit Karpenter’s voluntary disruptions, if no budget is set their is a default of 10% set. When calculating if a budget will block nodes from disruption, Karpenter checks if the number of nodes being deleted is greater than the number of allowed disruptions. Budgets take into consideration voluntary disruptions through expiration, drift, emptiness and consolidation. If there are multiple budgets defined in the NodePool, Karpenter will honour the most restrictive of the budgets.
+
+By applying a combination of disruptions budgets and Pod Disruptions Budgets (PDBs) you get both application and platform voluntary disruption controls, this can help you move towards continually operations to protect workload availability. You can learn more about Karpenter NodePool disruption budges and how the Kapenter disruption controllers works in the [Karpenter documentation](https://karpenter.sh/docs/concepts/disruption/#disruption-controller).
+
+## Examples
+The following provides a set of example disruption budgets:
+
+### Only interrupt a % of Karpenter managed nodes
+
+The following Disruption Budgets says, at any-point in time only disrupt 20% of the Nodes managed by the NodePool. For instance, if there were 19 nodes owned by the NodePool, 4 disruptions would be allowed, rounding up from 19 * .2 = 3.8.
+
+```
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  ...
+  disruption:
+    consolidationPolicy: WhenUnderutilized
+    expireAfter: 720h
+    budgets:
+    - nodes: "20%"
+```
+
+### Do not disrupt between UTC 09:00 and 18:00 every day
+
+The following Disruption Budgets says, for a 8 hour timeframe from UTC 9:00 don’t disrupt any nodes voluntary as this might be when the application is receiving peak traffic.
+
+```
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default-block-disruptions-during-standard-hours
+spec:
+  ...
+  disruption:
+    consolidationPolicy: WhenUnderutilized
+    expireAfter: 720h
+    budgets:
+    - nodes: "0"
+      schedule: "0 9 * * *"
+      duration: 8h
+```
+
+### Allow 20% disruptions during a maintenance window from UTC 22:00 to 2:00, but only 10% disruptions outside of a maintenance window
+
+The following Disruption Budgets says, for a 4 hour timeframe from UTC 22:00 only disrupt 20% of nodes, but during normal operations only disrupt 10% of nodes.
```

Comment:
When and why would you apply this configuration ?

Reply:
The use case I see for this is:

`You might apply this configuration if outside of core business hours you prefer a higher set of disruptions to speed up the role out of AMI updates for example.`